fsharp.extraction.targets

```diff
@@ -54,18 +54,18 @@
         </Task>
     </UsingTask>
 
-    <PropertyGroup>
-        <!-- Path where extracted fs files should be written by default -->
-        <FSTAR_EXTRACTED_PATH Condition=" '$(FSTAR_EXTRACTED_PATH)' == '' ">$(IntermediateOutputPath)extracted</FSTAR_EXTRACTED_PATH>
-        <!-- # 271: theory symbols in smt patters -->
-        <FSTAR_WARN_ERROR Condition=" '$(FSTAR_WARN_ERROR)' == '' ">--warn_error -271</FSTAR_WARN_ERROR>
-        <FSTAR_EXTRACTION_FLAGS Condition=" '$(FSTAR_EXTRACTION_FLAGS)' == '' ">--odir $(FSTAR_EXTRACTED_PATH) --codegen FSharp</FSTAR_EXTRACTION_FLAGS>
-        <!-- If FSTAR_HOME is not defined then assume that fstar.exe is available on the path -->
-        <FSTAR_EXE Condition="'$(FSTAR_HOME)' == ''">fstar.exe</FSTAR_EXE>
-        <FSTAR_EXE Condition="'$(FSTAR_HOME)' != ''">$(FSTAR_HOME)\bin\fstar.exe</FSTAR_EXE>
-    </PropertyGroup>
-
     <Target Name="GenerateFSharpSources" BeforeTargets="BeforeBuild;BeforeRebuild">
+
+        <PropertyGroup>
+            <!-- Path where extracted fs files should be written by default -->
+            <FSTAR_EXTRACTED_PATH Condition=" '$(FSTAR_EXTRACTED_PATH)' == '' ">$(IntermediateOutputPath)extracted</FSTAR_EXTRACTED_PATH>
+            <!-- # 271: theory symbols in smt patters -->
+            <FSTAR_WARN_ERROR Condition=" '$(FSTAR_WARN_ERROR)' == '' ">--warn_error -271</FSTAR_WARN_ERROR>
+            <FSTAR_EXTRACTION_FLAGS Condition=" '$(FSTAR_EXTRACTION_FLAGS)' == '' ">--odir $(FSTAR_EXTRACTED_PATH) --codegen FSharp</FSTAR_EXTRACTION_FLAGS>
+            <!-- If FSTAR_HOME is not defined then assume that fstar.exe is available on the path -->
+            <FSTAR_EXE Condition="'$(FSTAR_HOME)' == ''">fstar.exe</FSTAR_EXE>
+            <FSTAR_EXE Condition="'$(FSTAR_HOME)' != ''">$(FSTAR_HOME)\bin\fstar.exe</FSTAR_EXE>
```

Comment:
Thank you for this change!

I am actually in the process of revising how F* works for use in F# projects. 

I noticed that when trying to use this fsharp.extraction.targets file on Linux with a native fstar.exe executable (not .NET) that the build fails as shown below:


```
FStar/fsharp.extraction.targets(90,9): error MSB3073: The command "mono ../bin/fstar.exe --warn_error -271 --odir obj/Release/extracted --codegen FSharp --use_hints --lax --cache_checked_modules --cache_dir obj/Release/.cache.lax FStar.Pervasives.fst --extract "+FStar.Pervasives"" exited with code 2. 
```

Do you know what is causing msbuild to add "mono" as a prefix to the invocation of $(FSTAR_EXE)?

Reply:
Also pinging @mateuszbujalski who may know

Reply:
@nikswamy No idea. Could you add some console logging to see what the $(FSTAR_EXE) is set to? Are you using msbuild or xbuild on that machine? Perhaps it's some quirkiness in the implementation of Exec task on linux?

Reply:
Yeah, I did add some logging and the variable looks fine (no mono). But, somehow when the command is issued in the exec task `mono` is added as a prefix. I too suspect that it's a quirk of the Exec task on linux.

Reply:
can you add `/bl:fstar.binlog` and open file in the https://github.com/KirillOsenkov/MSBuildStructuredLog ? You can search for Exec task and copy exact command, then you can execute that command separately on your PC and maybe `fstar.exe` is bash wrapper somehow?